### PR TITLE
add value str,Enum

### LINKED ISSUE #2224: Enums for sprites broken with python 3.11+
**Describe the bug**
Use python3.11 or python3.12 and when you run the game you will get many errors like:

```
[2024-01-20 20:05:38,613] tuxemon.db - ERROR - validation failed for 'spyder_basement_roger': 1 validation error for NpcModel
template.0.sprite_name
  Value error, the sprite tennisplayer_lapi doesn't exist in the db [type=value_error, input_value='tennisplayer_lapi', input_type=str]
    For further information visit https://errors.pydantic.dev/2.5/v/value_error
[2024-01-20 20:05:38,613] tuxemon.db - ERROR - validation failed for 'spyder_basement_josh': 1 validation error for NpcModel
template.0.sprite_name
  Value error, the sprite boss_orange doesn't exist in the db [type=value_error, input_value='boss_orange', input_type=str]
    For further information visit https://errors.pydantic.dev/2.5/v/value_error
```

**To Reproduce**
1. Install python3.11+
2. Finish rest of installation
3. Run the game

**Expected behavior**
Sprites should load without these errors.

**Desktop (please complete the following information):**
 - OS: Mac M1 Sonoma 14.2.1
 - Python Version - 3.11 or 3.12

**Possible Solution**
Replace all enum usages with `.value`, like in npc.py:

```python
        for standing_type in list(EntityFacing):
            # if the template slug is interactive_obj, then it needs _front
            if self.interactive_obj:
                filename = f"{self.sprite_name}.png"
                path = os.path.join("sprites_obj", filename)
            else:
                filename = f"{self.sprite_name}_{standing_type}.png"
                ...
```

the line `filename = f"{self.sprite_name}_{standing_type}.png"` should instead be `filename = f"{self.sprite_name}_{standing_type.value}.png"`

### LINKED ISSUE #2239: Game freezes after first battle
**Describe the bug**
The game freezes after the first encounter in Cathedral mode.

**To Reproduce**
1. Run `python3 run_tuxemon.py` on fresh cloned repo on main after install the dependencies on venv.
> The output of this command here is:
> ```
> pygame-menu 4.4.3
> Git Hash: 034f5de1
> Enabling logging of all modules.
> [2024-01-31 13:54:20,447] tuxemon.rumble - ERROR - No rumble backends available.
> ```
2. Select 'New Game'
3. Select 'Tuxemon: Spyder and Cathedral'
4. Select any gender
5. Skip intro
6. Put any name
7.  Go through first dialog and select the name of any Tuxemon
8. Go down and right in the first map
9. Talk with shopkeeper
10. Select a Tux on bin
11. Fight the girl
12. After the fight the game freeze.

**Screenshots**
This is where the game become not responsive, there's no dialog and animations (e.g. flowers) and music are still going as normal.
image

**Desktop (please complete the following information):**
 - OS: Debian 12
 - Python Version: 3.11.2

**Additional context**
Inhalation method: Clone the git repo, create a virtual env and run the main python script on main branch.

### PR DESCRIPTION
**Pull request by JaskRendix**
Add **.value** to the **OutputBattle** to fix #2239 (fix #2224 already been fixed)

### CODE REVIEW

**Comment by CourteousCoder** [comment]
I'm taking a look now

File: tuxemon/combat.py [hunk 1]
```diff
@@ -406,7 +409,7 @@ def track_battles(
                 set_battle(session, OutputBattle.won, winner, _loser)
             # set variables
             if winner.isplayer:
-                set_var(session, "battle_last_result", OutputBattle.won)
+                set_var(session, "battle_last_result", _won)
```
Comment (CourteousCoder) [inline tuxemon/combat.py:412]: Winning the first battle works as expected. [Screenshot](https://i.imgur.com/18l6I1T.png)

File: tuxemon/combat.py [hunk 2]
```diff
@@ -434,11 +437,11 @@ def track_battles(
         loser = player
         winners = players
         info = {"name": loser.name.upper()}
-        set_var(session, "teleport_clinic", OutputBattle.lost)
+        set_var(session, "teleport_clinic", _lost)
```
Comment (CourteousCoder) [inline tuxemon/combat.py:440]: Losing the first battle works as expected. [Screenshot](https://i.imgur.com/ckCOqYN.png)

**Review by CourteousCoder — APPROVED** [review]
TLDR: I approve, and I think we should merge it.

I have tested and verified:
* Winning a trainer battle
* Losing a trainer battle
* Winning against a wild Tuxemon
* Running from a wild Tuxemon

I have not tested:
* Forfeiting from a trainer battle
* Ending a battle in a draw
* Losing against a wild Tuxemon

I still think this PR is good to go, because if there are issues with the untested cases, then it would be due to something unrelated to changing the way the values of a `str,Enum` are accessed.

**Comment by JaskRendix** [comment]
Sure, thanks for double checking, sooner or later it'll be merged

**Review by Grimmys — APPROVED** [review]

**Review by bitcraft — APPROVED** [review]
